docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,861 @@
+
+We will deploy three Control plane (master) nodes. Their Ignition configs can be create like so:
+
+```sh
+$ for index in $(seq 0 2); do
+    MASTER_HOSTNAME="$INFRA_ID-master-$index\n"
+    python3 -c "import base64, json, sys;
+try: b64encode = base64.encodebytes;
+except AttributeError: b64encode = base64.encodestring
+ignition = json.load(sys.stdin);
+files = ignition['storage'].get('files', []);
+files.append({'path': '/etc/hostname', 'mode': 420, 'contents': {'source': 'data:text/plain;charset=utf-8;base64,' + b64encode(b'$MASTER_HOSTNAME').decode().strip(), 'verification': {}}, 'filesystem': 'root'});
+ignition['storage']['files'] = files;
+json.dump(ignition, sys.stdout)" <master.ign >"$INFRA_ID-master-$index-ignition.json"
+done
+```
+
+This should create files `openshift-qlvwv-master-0-ignition.json`, `openshift-qlvwv-master-1-ignition.json` and `openshift-qlvwv-master-2-ignition.json`.
+
+If you look inside, you will see that they contain very little. In fact, most of the master Ignition is served by the Machine Config Server running on the bootstrap node and the masters contain only enough to know where to look for the rest.
+
+You can make your own changes here.
+
+**NOTE**: The worker nodes do not require any changes to their Ignition, but you can make your own by editing `worker.ign`.
+
+
+## Network Topology
+
+In this section we'll create all the networking pieces necessary to host the OpenShift cluster: network, subnet, router etc.
+
+
+### Network and Subnet
+
+We will use the `192.0.2.0/24` subnet range, but remove the first ten IP addresses from the allocation pool.
+
+These addresses will be used for the VRRP addresses managed by keepalived for high availability. For more information, read the [networking infrastructure design document][net-infra].
+
+[net-infra]: https://github.com/openshift/installer/blob/master/docs/design/openstack/networking-infrastructure.md
+
+```sh
+$ openstack network create "$INFRA_ID-network"
+$ openstack subnet create --subnet-range 192.0.2.0/24 --allocation-pool start=192.0.2.10,end=192.0.2.254 --network "$INFRA_ID-network" "$INFRA_ID-subnet"
+```
+
+### Subnet DNS (optional)
+
+During deployment, the OpenShift nodes will need to be able to resolve public name records to download the OpenShift images and so on. They will also need to resolve the OpenStack API endpoint.
+
+The default resolvers are an often set up by the OpenStack administrator in Neutron. However, some deployments do not have default DNS servers set, meaning the servers are not able to resolve any records when they boot.
+
+If you are in this situation, you can add resolvers to your Neutron subnet (`openshift-qlvwv-subnet`). These will be put into `/etc/resolv.conf` on your servers post-boot.
+
+For example, if you want to add the following nameservers: `198.51.100.86` and `198.51.100.87`, you can run this command:
+
+```sh
+$ openstack subnet --dns-nameserver 198.51.100.86 --dns-nameserver 198.51.100.87 "$INFRA_ID-subnet"
```

Comment:
This command should be `openstack subnet set`, I think.

Reply:
Indeed! Thanks for noticing.